volume/local/local.go

```diff
@@ -396,6 +400,33 @@ func (v *localVolume) LiveRestoreVolume(ctx context.Context, _ string) error {
 	return nil
 }
 
+// restoreIfMounted restores the mounted status if the _data directory is already mounted.
+func (v *localVolume) restoreIfMounted() {
+	if v.needsMount() {
+		// Check if the _data directory is already mounted.
+		// Always false on Windows.
+		mounted, err := mountinfo.Mounted(v.path)
+		if err != nil {
+			log.G(context.TODO()).WithFields(log.Fields{
+				"volume": v.name,
+				"path":   v.path,
+				"error":  err,
+			}).Debug("failed to determine if volume is already mounted, assuming no")
```

Comment:
Perhaps this should be a `Warn()`. This should not fail in normal circumstances, so probably better to make it slightly more noisy than "debug".

I'm also considering if we should return errors in this function, and make the caller log the warning (caller is already logging things), so change to something like;

For windows 😉 ;

```go
// restoreIfMounted restores the mounted status if the _data directory is already
// mounted. It is a no-op on Windows.
func (v *localVolume) restoreIfMounted() error {
	return nil
}
```

For unix;

```go
// restoreIfMounted restores the mounted status if the _data directory is already mounted.
func (v *localVolume) restoreIfMounted() error {
	// Check if the _data directory is already mounted.
	mounted, err := mountinfo.Mounted(v.path)
	if err != nil {
		return fmt.Errorf("failed to determine if volume is already mounted")
	}
	if mounted {
		// Mark volume as mounted, but don't increment active count. If
		// any container needs this, the refcount will be incremented
		// by the live-restore (if enabled).
		// In other case, refcount will be zero but the volume will
		// already be considered as mounted when Mount is called, and
		// only the refcount will be incremented.
		v.active.mounted = true
	}
	return nil
}
```


And call it as;

```go
if err := v.restoreIfMounted(); err != nil {
	log.G(context.TODO()).WithFields(log.Fields{
		"volume": v.name,
		"path":   v.path,
		"error":  err,
	}).Warn("failed to restoreIfMounted")
}
```

Reply:
Wondering in the above; do we need to set `v.active.mounted` to `false` as well, if we successfully determined that it was not mounted? Or would it already be `false` ?

Reply:
We are operating on a fresh `localVolume` struct here so `v.active.mounted` will already be false.